### PR TITLE
Feature 3 freights

### PR DESCRIPTION
**Pull request by kaeline**
https://github.com/Metatavu/vp-kuljetus-delivery-info-service-api/issues/3

### CODE REVIEW

**Comment by kaeline** [comment]
Note: Freight removal removes connected freight units

**Comment by kaeline** [comment]
DO not review

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/freights/freightunits/FreightUnitApiImpl.kt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    @Inject
+    lateinit var vertx: Vertx
+
+    @RolesAllowed(DRIVER_ROLE, MANAGER_ROLE)
+    override fun listFreightUnits(freightId: UUID?, first: Int?, max: Int?): Uni<Response> = CoroutineScope(vertx.dispatcher()).async {
+        val freightFilter = if (freightId != null) {
+        val freight = freightController.findFreight(freightId) ?: return@async createNotFound(createNotFoundMessage(FREIGHT, freightId))
+            freight
+        } else null
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/vp/deliveryinfo/freights/freightunits/FreightUnitApiImpl.kt:47]:
```suggestion
        val freightFilter = freightId?.let {
          freightController.findFreight(freightId) ?: return@async createNotFound(createNotFoundMessage(FREIGHT, freightId))
        }
```

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/freights/freightunits/FreightUnitApiImpl.kt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    @RolesAllowed(DRIVER_ROLE, MANAGER_ROLE)
+    @WithTransaction
+    override fun updateFreightUnit(freightUnitId: UUID, freightUnit: FreightUnit): Uni<Response> = CoroutineScope(vertx.dispatcher()).async {
+        val userId = loggedUserId ?: return@async createUnauthorized(UNAUTHORIZED)
+
+        val existingFreightUnit = freightUnitController.findFreightUnit(freightUnitId) ?: return@async createNotFound(
+            createNotFoundMessage(FREIGHT_UNIT, freightUnitId)
+        )
+        val newFreightUnit = if (existingFreightUnit.freight.id != freightUnit.freightId) {
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/vp/deliveryinfo/freights/freightunits/FreightUnitApiImpl.kt:79]: Isn't this `targetFreight` instead?

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/freights/freightunits/FreightUnitController.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    /**
+     * Updates a freight unit
+     *
+     * @param freightUnit freight unit
+     * @param newFreight new freight
+     * @param freightUnitRest rest model
+     * @param userId user id
+     * @return updated freight unit
+     */
+    suspend fun updateFreight(
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/vp/deliveryinfo/freights/freightunits/FreightUnitController.kt:73]:
```suggestion
    suspend fun updateFreightUnit(
```

**Review by villejuutila — APPROVED** [review]

**Pull request merged**